lib/elixir/lib/calendar/iso.ex

```diff
@@ -2009,34 +2007,42 @@ defmodule Calendar.ISO do
     {{0, 0}, rest}
   end
 
+  defp parse_microsecond(<<head, tail::binary>>, 6, acc) when head in ?0..?9,
+    do: parse_microsecond(tail, 6, acc)
+
```

Comment:
when there is more than 6 digits of milicseconds we just drop the head and continue.
It does not reequire special handling in the caller